lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -52,26 +52,27 @@
        configurations conforms to task arguments.'
     - 'If C(state) is set to C(absent) and virtual machine exists, then the specified virtual machine
       is removed with its associated components.'
-    - 'If C(state) is set to one of the following C(poweredon), C(poweredoff), C(present), C(restarted), C(suspended)
-      and virtual machine does not exists, then virtual machine is deployed with given parameters.'
-    - 'If C(state) is set to C(poweredon) and virtual machine exists with powerstate other than powered on,
+    - 'If C(state) is set to one of the following C(poweredon), C(powered-on), C(poweredoff), C(powered-off),
+      C(present), C(restarted), C(suspended) and virtual machine does not exists, virtual machine is deployed with the given parameters.'
+    - 'If C(state) is set to C(poweredon) or C(powered-on) and virtual machine exists with powerstate other than powered on,
```

Comment:
In order to keep things simple and short, I would prefer to just mention the new state names, and add a comment to list the aliases. e.g: `poweredoff` is an alias for `powered-off`.
I also would like to see a deprecation warning when the old state are used. This way, we won't have to keep all the state names for decades.